src/main/java/org/kuali/kra/irb/actions/ProtocolProtocolActionsAction.java

```diff
@@ -2275,11 +2275,16 @@ public ActionForward viewActionCorrespondence(ActionMapping mapping, ActionForm
         int actionIndex = getSelectedLine(request);
         int attachmentIndex = getSelectedAttachment(request);
         org.kuali.kra.irb.actions.ProtocolAction protocolAction = (org.kuali.kra.irb.actions.ProtocolAction) protocolForm.getActionHelper().getProtocol().getProtocolActions().get(actionIndex);
-        ProtocolCorrespondence attachment = (ProtocolCorrespondence) protocolAction.getProtocolCorrespondences().get(attachmentIndex);
+        ProtocolCorrespondence attachment = null;
+
+        if (protocolAction.getProtocolCorrespondences().size() > 0) {
```

Comment:
The check on the size also needs to see if size() > attachmentIndex otherwise it is possible to get an index out of bounds exception.